src/colmap/geometry/triangulation.cc

```diff
@@ -37,63 +37,58 @@
 
 namespace colmap {
 
-Eigen::Vector3d TriangulatePoint(const Eigen::Matrix3x4d& cam1_from_world,
-                                 const Eigen::Matrix3x4d& cam2_from_world,
-                                 const Eigen::Vector2d& point1,
-                                 const Eigen::Vector2d& point2) {
-  Eigen::Matrix4d A;
+bool TriangulatePoint(const Eigen::Matrix3x4d& cam1_from_world,
+                      const Eigen::Matrix3x4d& cam2_from_world,
+                      const Eigen::Vector2d& point1,
+                      const Eigen::Vector2d& point2,
+                      Eigen::Vector3d* xyz) {
+  THROW_CHECK_NOTNULL(xyz);
 
+  Eigen::Matrix4d A;
   A.row(0) = point1(0) * cam1_from_world.row(2) - cam1_from_world.row(0);
   A.row(1) = point1(1) * cam1_from_world.row(2) - cam1_from_world.row(1);
   A.row(2) = point2(0) * cam2_from_world.row(2) - cam2_from_world.row(0);
   A.row(3) = point2(1) * cam2_from_world.row(2) - cam2_from_world.row(1);
 
-  Eigen::JacobiSVD<Eigen::Matrix4d> svd(A, Eigen::ComputeFullV);
-
-  return svd.matrixV().col(3).hnormalized();
-}
-
-std::vector<Eigen::Vector3d> TriangulatePoints(
-    const Eigen::Matrix3x4d& cam1_from_world,
-    const Eigen::Matrix3x4d& cam2_from_world,
-    const std::vector<Eigen::Vector2d>& points1,
-    const std::vector<Eigen::Vector2d>& points2) {
-  THROW_CHECK_EQ(points1.size(), points2.size());
-
-  std::vector<Eigen::Vector3d> points3D(points1.size());
-
-  for (size_t i = 0; i < points3D.size(); ++i) {
-    points3D[i] = TriangulatePoint(
-        cam1_from_world, cam2_from_world, points1[i], points2[i]);
+  const Eigen::JacobiSVD<Eigen::Matrix4d> svd(A, Eigen::ComputeFullV);
+  if (svd.info() != Eigen::Success || svd.matrixV()(3, 3) == 0) {
```

Comment:
info() is only available in Eigen >= 3.4.0: https://gitlab.com/libeigen/eigen/-/commit/5bbc9cea93ef29cee2b8ffb2084d4ebca32600ba

Reply:
Thanks, didn't know. Will fix.